src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/Utf8BufferTextWriter.cs

```diff
@@ -77,12 +76,12 @@ public void SetWriter(IBufferWriter<byte> bufferWriter)
 
         public override void Write(char[] buffer, int index, int count)
         {
-            WriteInternal(buffer, index, count);
+            WriteInternal(buffer.AsSpan(index, count));
```

Comment:
Is this going to cause some perf hit on environments where span isn't fully optimized? @davidfowl

Reply:
We were talking about this just recently ;). We `#ifdef` and get a pointer from the span which mostly resolves any perf hit on slower platforms.

Reply:
Creating the span is slightly slower but we do pin it to avoid some of the costs.